lib/_stream_writable.js

```diff
@@ -658,11 +657,9 @@ function onCorkedFinish(corkReq, state, err) {
     cb(err);
     entry = entry.next;
   }
-  if (state.corkedRequestsFree) {
-    state.corkedRequestsFree.next = corkReq;
-  } else {
-    state.corkedRequestsFree = corkReq;
-  }
+
+  // reuse the free corkReq.
+  state.corkedRequestsFree.next = corkReq;
```

Comment:
Unfortunately, this can end up being `null` because it copied over in: https://github.com/MoonBall/node/blob/dd13a79b830ef2d33ac0288e197a431a63e40941/lib/_stream_writable.js#L517. Can you revert this?

Reply:
`if (holder.next)` should ensure that `state.corkedRequestsFree` is not null.

Reply:
My bad. I'm __very__ scared of https://github.com/nodejs/node/pull/6164 and https://github.com/nodejs/node/issues/6154